### PR TITLE
Retry on any ConnectionError exception

### LINKED ISSUE #47: `Name or service not known` errors are not retried
The `_hc_retry_on_exception` function only retries `ConnectionError` when the underlying exception is `BadStatusLine`. In this case, we got a `socket.gaierror('Name or service not known')`. See the full traceback below:

```
ConnectionError: ('Connection aborted.', gaierror(-2, 'Name or service not known'))
  File "sh_scrapy/crawl.py", line 127, in _run_usercode
    _run(args, settings)
  File "sh_scrapy/crawl.py", line 89, in _run
    _run_pkgscript(args)
  File "sh_scrapy/crawl.py", line 110, in _run_pkgscript
    d.run_script(scriptname, {'__name__': '__main__'})
  File "dist-packages/pkg_resources.py", line 1235, in run_script
    execfile(script_filename, namespace, namespace)
  File "scrapy/tmp/eggs-cJgpCT/__main__.egg/EGG-INFO/scripts/run-script.py", line 55, in <module>
    maybe_capture_errors(main, settings)(settings)
  File "usr/lib/python2.7/dist-packages/raven/base.py", line 681, in wrapper
    return func(*funcargs, **funckwargs)
  File "scrapy/tmp/eggs-cJgpCT/__main__.egg/EGG-INFO/scripts/run-script.py", line 50, in main
    target(**kwargs)
  File "ompostprocess/cli.py", line 192, in main
    spider_name = hs_job.metadata['spider']
  File "hubstorage/resourcetype.py", line 202, in __getitem__
    return self._data[key]
  File "hubstorage/resourcetype.py", line 178, in _data
    r = self.apiget()
  File "hubstorage/resourcetype.py", line 39, in apiget
    return self.apirequest(_path, method='GET', **kwargs)
  File "hubstorage/resourcetype.py" line 32, in apirequest
    return jldecode(self._iter_lines(_path, **kwargs))
  File "hubstorage/resourcetype.py", line 27, in _iter_lines
    r = self.client.request(**kwargs)
  File "hubstorage/client.py", line 100, in request
    return self.retrier.call(invoke_request)
  File "dist-packages/retrying.py", line 206, in call
    return attempt.get(self._wrap_exception)
  File "dist-packages/retrying.py", line 247, in get
    six.reraise(self.value[0], self.value[1], self.value[2])
  File "dist-packages/retrying.py", line 200, in call
    attempt = Attempt(fn(*args, **kwargs), attempt_number, False)
  File "hubstorage/client.py", line 92, in invoke_request
    r = self.session.request(**kwargs)
  File "usr/lib/python2.7/dist-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "usr/lib/python2.7/dist-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "usr/lib/python2.7/dist-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)

```

### PR DESCRIPTION
**Pull request by lsenta**
- Fixes #47 / SC-404

So far we checked ConnectionError causes, which appears unnecessary, we can just catch every ConnectionError, no matter the cause.

ConnectionError may be caused by:
ReadTimeoutError, ClosedPoolError, ProtocolError, socket.error

It may also be caused by specific cases of MaxRetryError that we won't encounter.

Signed-off-by: lsenta <laurent.senta@gmail.com>

### CODE REVIEW

**Comment by lsenta** [comment]
(recreating to trigger Jira integration)

**Comment by lsenta** [comment]
Assigning @immerrr since we had the previous talk on retrying

File: hubstorage/client.py [hunk 1]
```diff
@@ -28,10 +28,8 @@ def _hc_retry_on_exception(err):
         logger.warning("Server failed with %d status code, retrying (maybe)" % (err.response.status_code,))
         return True
 
-    # TODO: python3 compatibility: BadStatusLine error are wrapped differently
-    if (isinstance(err, ConnectionError) and err.args[0] == 'Connection aborted.' and
-            isinstance(err.args[1], BadStatusLine) and err.args[1][0] == repr('')):
-        logger.warning("Protocol failed with BadStatusLine, retrying (maybe)")
+    if isinstance(err, ConnectionError):
+        logger.warning("Request encountered a connection error: %s, retrying (maybe)" % (err,))
```
Comment (immerrr) [inline hubstorage/client.py]: Please use `%r` to format the error to retain error type information, as str only returns the message.

Also, logging module can format error messages, like `logger.warning("... %r ...", err)`, and postpone the formatting until it's actually needed. I see that has been done above, so might fix both cases while you're at it.

**Comment by immerrr** [comment]
one minor nitpick, aside from that, :+1:

**Comment by immerrr** [comment]
lgtm now

**Pull request merged**